tests/test_2025_07_25_hoodi_dg_upgrade.py

```diff
@@ -0,0 +1,320 @@
+
+    # =======================================================================
+    # ================= After DG proposal execution tests ===================
+    # =======================================================================
+
+    assert timelock.getGovernance() == NEW_DUAL_GOVERNANCE
+    assert dual_governance.getConfigProvider() == CONFIG_PROVIDER_FOR_DISCONNECTED_DUAL_GOVERNANCE
+
+    assert new_dual_governance.getConfigProvider() == ACTIVE_CONFIG_PROVIDER
+    assert new_dual_governance.getResealCommittee() == RESEAL_COMMITTEE
+    assert new_dual_governance.getProposalsCanceller() == VOTING
+    proposers = new_dual_governance.getProposers()
+    assert len(proposers) == 1
+    assert proposers[0][0] == VOTING
+    assert proposers[0][1] == ADMIN_EXECUTOR
+    assert new_dual_governance.getTiebreakerDetails() == (
+        False,
+        NEW_TIEBREAKER_COMMITTEE,
+        TIEBREAKER_ACTIVATION_TIMEOUT,
+        [WITHDRAWAL_QUEUE, VALIDATORS_EXIT_BUS_ORACLE, TRIGGERABLE_WITHDRAWALS_GATEWAY],
+    )
+
+    # Test that old DG can't submit proposals
+    with reverts("CallerIsNotGovernance: 0x4d12b9f6acab54ff6a3a776ba3b8724d9b77845f"):
+        dual_governance.submitProposal([], "This should revert", {"from": VOTING})
+
+    # Test that old escrow won't get into Rage Quit if lock amount above previous 2nd seal
+    old_escrow = interface.DualGovernanceEscrow(contracts.dual_governance.getVetoSignallingEscrow())
+    new_escrow = interface.DualGovernanceEscrow(new_dual_governance.getVetoSignallingEscrow())
+
+    old_dual_governance_second_seal = interface.IDualGovernanceConfigProvider(
+        dual_governance.getConfigProvider()
+    ).SECOND_SEAL_RAGE_QUIT_SUPPORT()
+    new_dual_governance_second_seal = interface.IDualGovernanceConfigProvider(
+        new_dual_governance.getConfigProvider()
+    ).SECOND_SEAL_RAGE_QUIT_SUPPORT()
+
+    steth_total_supply = steth_token.totalSupply()
+
+    # 20% steth of total supply is greater than previous 2nd seal - 10%
+    steth_to_lock = steth_total_supply * 0.2
+
+    steth_whale = accounts.at("0xF865A1d43D36c713B4DA085f32b7d1e9739B9275", force=True)
+    steth_token.approve(old_escrow.address, steth_to_lock, {"from": steth_whale})
+
+    old_dual_governance_rage_quit_support_before = old_escrow.getRageQuitSupport()
+    new_dual_governance_rage_quit_support_before = new_escrow.getRageQuitSupport()
+
+    old_escrow.lockStETH(steth_to_lock, {"from": steth_whale})
+
+    assert old_escrow.getRageQuitSupport() > old_dual_governance_rage_quit_support_before
+    assert new_escrow.getRageQuitSupport() == new_dual_governance_rage_quit_support_before
+
+    assert old_escrow.getRageQuitSupport() < old_dual_governance_second_seal
+    assert old_escrow.getRageQuitSupport() > new_dual_governance_second_seal
+
+    chain.sleep(old_escrow.MAX_MIN_ASSETS_LOCK_DURATION() + 1)
+    old_escrow.unlockStETH({"from": steth_whale})
```

Comment:
Maybe it's worth adding lock and unlock for the new escrow to check that it's connected in correct way and has effect on the new dg state.